src/shared/loop-util.c

```diff
@@ -706,9 +708,21 @@ LoopDevice* loop_device_unref(LoopDevice *d) {
                         log_debug_errno(errno, "Failed to sync loop block device, ignoring: %m");
 
                 if (d->nr >= 0 && !d->relinquished) {
```

Comment:
BTW, not a matter, but why this code block is conditionalized with `d->nr >= 0`, instead of `d->fd >= 0`??

Reply:
to simplify things like `systemd-nspawn --image=/dev/sdc` we actually allow LoopDevice objects to be created from non-loopback devices too. i.e. if you point loop_device_make() to an existing regular block device, then we'll just use that as is, and not create a loopback device from the thing that is anyway a block device.

`d->nr` is set to the loopback device index (i.e. nr=4 for /dev/loop4), but when wrapping a block device that actually is not really a loopback bock device, we set it to -1.

I will add some comments about this.

Reply:
(and when we wrap such a non-loopback block device, then we of course don't have to release it, it should just stay around, after all it's not managed by us then)

Reply:
Oh, so LoopDevice may not be related to a loopback block device... Ugh.